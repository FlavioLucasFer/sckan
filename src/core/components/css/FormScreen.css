.screen-form {
	overflow-y: auto;
}

.buttons-row {
	display: flex;
	justify-content: flex-end;
	align-items: flex-end;

	width: 100%;
	height: 100%;
	margin: 0 !important;
}