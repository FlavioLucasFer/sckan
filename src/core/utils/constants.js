const MAIN_COLOR = '#1ac3d6';
const SECUNDARY_COLOR = '#245257';

export {
    MAIN_COLOR,
    SECUNDARY_COLOR,
};