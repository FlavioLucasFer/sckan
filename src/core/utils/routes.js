const HOME_ROUTE = '/';
const LOGIN_ROUTE = '/login';
const COMPANY_ROUTE = '/companies';
const COMPANY_FORM_ROUTE = `${COMPANY_ROUTE}/form`;
const PROJECT_ROUTE = '/projects';
const PROJECT_FORM_ROUTE = `${PROJECT_ROUTE}/form`;
const SPRINT_ROUTE = '/sprints';
const SPRINT_FORM_ROUTE = `${SPRINT_ROUTE}/form`;
const TASK_ROUTE = '/tasks';
const TASK_FORM_ROUTE = `${TASK_ROUTE}/form`;
const USER_ROUTE = '/users';
const USER_FORM_ROUTE = `${USER_ROUTE}/form`;
const USER_PROFILE = '/user';
const USER_SITTINGS = `${USER_PROFILE}/settings`;

export {
	HOME_ROUTE,
  LOGIN_ROUTE,
  COMPANY_ROUTE,
	COMPANY_FORM_ROUTE,
  PROJECT_ROUTE,
	PROJECT_FORM_ROUTE,
  SPRINT_ROUTE,
	SPRINT_FORM_ROUTE,
  TASK_ROUTE,
	TASK_FORM_ROUTE,
  USER_ROUTE,
	USER_FORM_ROUTE,
	USER_PROFILE,
	USER_SITTINGS,
};