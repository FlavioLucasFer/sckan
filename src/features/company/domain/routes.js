'use strict';

const { REACT_APP_API_URL } = process.env;

const BASE_URL = `${REACT_APP_API_URL}/companies`;

const ROUTES = {
	CREATE: `${BASE_URL}/create`,
	EDIT: `${BASE_URL}/edit`,
	FIND: `${BASE_URL}/find`,
	FIND_BY_ID: `${BASE_URL}/find/:id`,
	REMOVE: `${BASE_URL}/remove`,
};

export default ROUTES;
