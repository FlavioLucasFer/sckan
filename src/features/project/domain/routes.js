'use strict'

const { REACT_APP_API_URL } = process.env;

const BASE_URL = `${REACT_APP_API_URL}/projects`;

const ROUTES = {
	CREATE: `${BASE_URL}/create`,
	EDIT: `${BASE_URL}/edit`,
	REMOVE: `${BASE_URL}/remove`,
	FIND: `${BASE_URL}/find`,
};

export default ROUTES;
