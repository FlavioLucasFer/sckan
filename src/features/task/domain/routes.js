'use strict'

const { REACT_APP_API_URL } = process.env;

const BASE_URL = `${REACT_APP_API_URL}/tasks`;

const ROUTES = {
	CREATE: `${BASE_URL}/create`,
	EDIT: `${BASE_URL}/edit`,
	REMOVE: `${BASE_URL}/remove`,
	FIND: `${BASE_URL}/find`,
	FIND_BY_ID: `${BASE_URL}/find/:id`,
};

export default ROUTES;
