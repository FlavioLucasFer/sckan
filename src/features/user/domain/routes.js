'use strict'

const { REACT_APP_API_URL } = process.env;

const BASE_URL = `${REACT_APP_API_URL}/users`;

const ROUTES = {
	CREATE: `${BASE_URL}/create`,
	EDIT: `${BASE_URL}/edit`,
	REMOVE: `${BASE_URL}/remove`,
	FIND: `${BASE_URL}/find`,
	FIND_BY_ID: `${BASE_URL}/find/:id`,
	SIGN_IN: `${BASE_URL}/sign-in`,
	LOGOUT: `${BASE_URL}/logout`,
};

export default ROUTES;
