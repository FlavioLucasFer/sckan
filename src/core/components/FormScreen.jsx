import './css/FormScreen.css';
import React from 'react';
import PropTypes from 'prop-types';
import { useDispatch } from 'react-redux';
import { useHistory } from 'react-router-dom';

import ConfirmActionModal from 'core/modals/ConfirmActionModal';
import CardPanel from 'core/components/CardPanel';
import TextInput from 'core/components/TextInput';
import Button from 'core/components/Button';

import { setCurrentScreen } from 'redux/reducers/currentScreen';

FormScreen.propTypes = {
	title: PropTypes.string.isRequired,
	backRoute: PropTypes.string.isRequired,
	id: PropTypes.string,
	children: PropTypes.element,
	onSave: PropTypes.func,
}

function FormScreen(props) {
	const { 
		title,
		backRoute,
		id,
		onSave,
		children,
	} = props;

	const dispatch = useDispatch();
	const { push } = useHistory();

	async function handleSave() {
		if (onSave) {
			try {
				await onSave();

				goBack();
			} catch (err) {
				throw err;
			}
		} else {
			goBack();
		}
	}

	function goBack() {
		push(backRoute);
		dispatch(setCurrentScreen(backRoute));
	}

	return (
		<section>
			<h2 className="screen-title grey-text text-darken-3">
				{id ? `Edit ${title.toLowerCase()}` : `New ${title.toLowerCase()}`}
			</h2>

			<CardPanel className="s12 div-card-container"
				cardClassName="card-container">
				<div className="row" style={{ height: '100%' }}>
					<form className="col s12 screen-form" style={{ height: '90%' }}>
						{id &&
							<div className="col s12 no-padding">
								<TextInput value={`#${id}`}
									className="col s4 no-padding"
									inputClassName="bold-text grey-text text-darken-2"
									disabled />
							</div>
						}
						{children}
					</form>
					<div className="col s12" style={{ height: '10%' }} >
						<div className="row buttons-row">
							<a className="modal-trigger" 
								href={id && '#confirm-save-modal'}
								onClick={!id && handleSave}>
								<Button title="Save"
									className="bold-text"
									color="teal accent-4"
									icon={{
										name: 'done',
										className: 'right',
									}} />
							</a>
							<a className="modal-trigger" 
								href="#confirm-discard-modal"
								style={{ marginLeft: '10px' }}>
								<Button title="Discard"
									className="bold-text"
									color="red accent-2"
									icon={{
										name: 'cancel',
										className: 'right',
									}} />
							</a>
						</div>
					</div>
				</div>
			</CardPanel>

			<ConfirmActionModal id="confirm-save-modal"
				title="Are you sure you want to save changes?"
				onConfirm={handleSave} />
			<ConfirmActionModal id="confirm-discard-modal"
				title="Are you sure you want to discard changes?"
				onConfirm={goBack} />
		</section>
	);
}

export default FormScreen;