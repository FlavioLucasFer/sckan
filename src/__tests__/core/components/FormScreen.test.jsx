import React from 'react';
import { shallow } from 'enzyme';

import FormScreen from 'core/components/FormScreen';
import { COMPANY_FORM_ROUTE, COMPANY_ROUTE } from 'core/utils/routes';

const mockHistory = {
	goBack: jest.fn(),
};

jest.mock('react-router-dom', () => ({
	...(jest.requireActual('react-router-dom')),
	useHistory: () => (mockHistory),
}));

describe('Testing FormScreen core component', () => {
	afterAll(() => {
		jest.resetAllMocks();
	});

	function testGoBackCall(modalId, buttonId, times) {
		const wrapper = shallow(
			<FormScreen title="testing goBack">
				<div>
					<h3>test</h3>
				</div>
			</FormScreen>
		);

		wrapper.find({ id: modalId })
			.dive()
			.find({ id: modalId })
			.dive()
			.find({ id: buttonId })
			.simulate('click');

		expect(mockHistory.goBack).toBeCalledTimes(times);
	}
	
	it('should render correctly', () => {
		const wrapper = shallow(
			<FormScreen title="form screen test">
				<h2>testing</h2>
			</FormScreen>
		);

		expect(wrapper.debug()).toMatchSnapshot();

		wrapper.setProps({
			title: 'testing',
			id: 1818,
		});

		expect(wrapper.debug()).toMatchSnapshot();
	});

	it('should call goBack function only 1 time by confirm save', () => {
		testGoBackCall('confirm-save-modal', 'confirm-action', 1);
	});

	it('should call goBack function only 1 time by confirm discard', () => {
		testGoBackCall('confirm-discard-modal', 'confirm-action', 1);
	});

	it('should not call goBack function by deny save', () => {
		testGoBackCall('confirm-save-modal', 'deny-action', 0);
	});

	it('should not call goBack function by deny discard', () => {
		testGoBackCall('confirm-discard-modal', 'deny-action', 0);
	});
});