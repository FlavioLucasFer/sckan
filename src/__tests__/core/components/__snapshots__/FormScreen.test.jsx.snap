// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing FormScreen core component should render correctly 1`] = `
"<section>
  <h2 className=\\"screen-title grey-text text-darken-3\\">
    New form screen test
  </h2>
  <CardPanel className=\\"s12 div-card-container\\" cardClassName=\\"card-container\\">
    <div className=\\"row\\" style={{...}}>
      <form className=\\"col s12 screen-form\\" style={{...}}>
        <h2>
          testing
        </h2>
      </form>
      <div className=\\"col s12\\" style={{...}}>
        <div className=\\"row buttons-row\\">
          <a className=\\"modal-trigger\\" href={[undefined]} onClick={[Function: handleSave]}>
            <Button title=\\"Save\\" className=\\"bold-text\\" color=\\"teal accent-4\\" icon={{...}} />
          </a>
          <a className=\\"modal-trigger\\" href=\\"#confirm-discard-modal\\" style={{...}}>
            <Button title=\\"Discard\\" className=\\"bold-text\\" color=\\"red accent-2\\" icon={{...}} />
          </a>
        </div>
      </div>
    </div>
  </CardPanel>
  <ConfirmActionModal id=\\"confirm-save-modal\\" title=\\"Are you sure you want to save changes?\\" onConfirm={[Function: handleSave]} />
  <ConfirmActionModal id=\\"confirm-discard-modal\\" title=\\"Are you sure you want to discard changes?\\" onConfirm={[Function: handleDiscard]} />
</section>"
`;

exports[`Testing FormScreen core component should render correctly 2`] = `
"<section>
  <h2 className=\\"screen-title grey-text text-darken-3\\">
    Edit testing
  </h2>
  <CardPanel className=\\"s12 div-card-container\\" cardClassName=\\"card-container\\">
    <div className=\\"row\\" style={{...}}>
      <form className=\\"col s12 screen-form\\" style={{...}}>
        <div className=\\"col s12 no-padding\\">
          <TextInput value=\\"#1818\\" className=\\"col s4 no-padding\\" inputClassName=\\"bold-text grey-text text-darken-2\\" disabled={true} />
        </div>
        <h2>
          testing
        </h2>
      </form>
      <div className=\\"col s12\\" style={{...}}>
        <div className=\\"row buttons-row\\">
          <a className=\\"modal-trigger\\" href=\\"#confirm-save-modal\\" onClick={false}>
            <Button title=\\"Save\\" className=\\"bold-text\\" color=\\"teal accent-4\\" icon={{...}} />
          </a>
          <a className=\\"modal-trigger\\" href=\\"#confirm-discard-modal\\" style={{...}}>
            <Button title=\\"Discard\\" className=\\"bold-text\\" color=\\"red accent-2\\" icon={{...}} />
          </a>
        </div>
      </div>
    </div>
  </CardPanel>
  <ConfirmActionModal id=\\"confirm-save-modal\\" title=\\"Are you sure you want to save changes?\\" onConfirm={[Function: handleSave]} />
  <ConfirmActionModal id=\\"confirm-discard-modal\\" title=\\"Are you sure you want to discard changes?\\" onConfirm={[Function: handleDiscard]} />
</section>"
`;
